### DISCUSSION TITLE
Custom extensibility library not loading

### DISCUSSION THREAD

**Discussion by phoob** [microsoft-search/pnp-modern-search#1693] [Q&A] [answered]
Hello! I am posting this here since I have no idea if this is a bug or if I am just doing it wrong. 

I have tried creating a custom extensibility library according to the documentation, with a few "hello world" additions. But when I try to enable it in a search results web part, it just doesn't seem to load.

## What I did:

1. upload the built library package to the tenants application catalog and deployed it tenant-wide (i also tried just using `gulp serve` without uploading it with the same result)
2. created a page with a search results web part
3. added the id for the libary to the web parts extensibility configuration and activated it
4. saved and reloaded the page

## Result:

I've tried both adding a (simple) custom component, a handlebars helper and a custom layout – but none of them seem to load.

I even tried throwing in a console.log statement in the main file, but even that is not printed.

What am I doing wrong?

**Comment by FranckyC** [comment, marked as answer]
Hi @phoob, did you reuse the sample library from the https://github.com/microsoft-search/pnp-modern-search-extensibility-samples repository?

**Reply by FranckyC** [reply]
A common mistake is to forget the "deploy to all sites" flag at deployment time when uploading the package.

**Reply by phoob** [reply]
No, I created a new library according to the docs here: https://microsoft-search.github.io/pnp-modern-search/extensibility/#create-an-extensibility-library , and Yes, I deployed it to all sites.

**Reply by phoob** [reply]
Btw. I have also tried disabling the default extensibility library – and that does not make any difference.

![image](https://user-images.githubusercontent.com/1475800/148548060-0556d079-223c-41b3-a125-ea17d1fc87c5.png)

![image](https://user-images.githubusercontent.com/1475800/148548135-c66341ba-9492-4637-9157-fcde0f538ad2.png)

**Reply by FranckyC** [reply]
Can you try with the sample library? Also, any error messages in the console or network tab?

**Reply by phoob** [reply]
Hi! I made it work with the sample library. Must have done something else wrong. Thanks! 

Also: I now realize that I should use that as a starting point, since it then will be loaded automatically by all pnp web parts. Awesome. Hadn't realized that!
